src/frontend/context/OppgaverContext.tsx

```diff
@@ -1,29 +1,129 @@
     const [oppgaver, settOppgaver] = React.useState<Ressurs<IHentOppgaveDto>>(
         byggTomRessurs<IHentOppgaveDto>()
     );
+    const [oppgaveFelter, settOppgaveFelter] = useState<IOppgaveFelter>(
+        initialOppgaveFelter(innloggetSaksbehandler, history.location.search)
+    );
+
+    useEffect(() => {
+        settOppgaveFelter(initialOppgaveFelter(innloggetSaksbehandler, history.location.search));
+    }, [innloggetSaksbehandler]);
+
+    useEffect(() => {
+        if (hentOppgaverVedSidelast && innloggetSaksbehandler) {
+            if (
+                Object.values(oppgaveFelter).filter(
+                    (oppgaveFelt: IOppgaveFelt) =>
+                        oppgaveFelt.filter?.initialValue !== oppgaveFelt.filter?.selectedValue
+                ).length > 0
+            ) {
+                hentOppgaver();
+            }
+            settHentOppgaverVedSidelast(false);
+        }
+    }, [oppgaveFelter]);
+
+    const hentOppgaveFelt = (nøkkel: string) => {
+        return oppgaveFelter[nøkkel];
+    };
+
+    const settVerdiPåOppgaveFelt = (oppgaveFelt: IOppgaveFelt, nyVerdi: string) => {
+        if (oppgaveFelt.filter) {
+            const oppdaterteOppgaveFelter = {
+                ...oppgaveFelter,
+                [oppgaveFelt.nøkkel]: {
+                    ...oppgaveFelt,
+                    filter: {
+                        ...oppgaveFelt.filter,
+                        selectedValue: nyVerdi,
+                    },
+                },
+            };
+            settOppgaveFelter(oppdaterteOppgaveFelter);
+
+            history.push({
+                search: Object.values(oppdaterteOppgaveFelter)
+                    .filter(
+                        (mapOppgaveFelt: IOppgaveFelt) =>
+                            mapOppgaveFelt.filter?.selectedValue !==
+                            mapOppgaveFelt.filter?.initialValue
+                    )
+                    .map((mapOppgaveFelt: IOppgaveFelt) => {
+                        return `${mapOppgaveFelt.nøkkel}=${mapOppgaveFelt?.filter?.selectedValue}`;
+                    })
+                    .join('&'),
+            });
+        }
+    };
+
+    const settSortOrderPåOppgaveFelt = (felt: string) => {
+        let midlertidigOppgaveFelter = oppgaveFelter;
+        Object.values(oppgaveFelter).forEach((oppgaveFelt: IOppgaveFelt) => {
+            if (oppgaveFelt.nøkkel === felt) {
+                midlertidigOppgaveFelter = {
+                    ...midlertidigOppgaveFelter,
+                    [oppgaveFelt.nøkkel]: {
+                        ...oppgaveFelt,
+                        order:
+                            oppgaveFelt.order === FeltSortOrder.ASCENDANT
+                                ? FeltSortOrder.DESCENDANT
+                                : FeltSortOrder.ASCENDANT,
+                    },
+                };
+            } else if (oppgaveFelt.order && oppgaveFelt.order !== FeltSortOrder.NONE) {
```

Comment:
If you clear the order on all other fields, the sort result is only based on the current toggled field. Is this a good behaviour?

Reply:
But does the sort method support multisort columns?